app/src/main/java/com/example/coronadiagnosticapp/ui/fragments/camera/CameraFragment.kt

```diff
@@ -59,35 +62,37 @@ class CameraFragment : ScopedFragment() {
         return inflater.inflate(R.layout.camera_fragment, container, false)
     }
 
+    private fun openOximeter() {
+        // TODO: Ask permission to use the camera
```

Comment:
It looks like it's already happening

Reply:
If I'm following the logic, it seems like you're
3. starting an Activity, 
2. from a fragment,
1. if a certain condition applies.

Seems like it is already half way decoupled. 
You have:
The activity for the actual detection process.
The fragment for the explanation thing.

I'd gather the first run parameter before I decide on whether I need to start the explanation fragment or the detection activity.

btw: I think that any solution should maintain the stepper for the oximeter stage. 
I would have considered my current solution with that in mind, which probably means getting rid of the oximeter activity and replacing it with a fragment that would maintain a consistent flow and feel.

Reply:
your right tho as i wrote at the top, this will take more time. I'm adding an issue for this.